clusterlib/tests/test_scheduler.py

```diff
@@ -105,7 +161,8 @@ def test_queued_or_running_jobs_slurm():
         assert_in(job_name, running_jobs)
     finally:
         # Make sure to clean up even if there is a failure
-        os.system("scancel %s" % job_id)
+        if job_id is not None:
+            os.system("scancel %s" % job_id)
```

Comment:
`job_id` is presumably not injectable by the user by a `subprocess.call(['scancel', job_id], shell=True)` would also be safer here.